test_compileall_original.py

```diff
@@ -352,6 +352,27 @@ def test_multiple_optimization_levels(self):
                 except Exception:
                     pass
 
+    def test_ignore_symlink_destination(self):
+        # Create folders for allowed files, symlinks and prohubited area
+        types = {"allowed": None, "symlinks": None, "prohibited": None}
+        for type in types:
+            folders = ["test", "dir", type]
+            types[type] = os.path.join(self.directory, *folders)
+            os.makedirs(types[type])
+
+        # Create script and remember their byte-compiled versions
+        bc = {"allowed": None, "prohibited": None}
+        for type in bc:
+            script = script_helper.make_script(types[type], "test_" + type, "a = 0")
+            symlink = os.path.join(types["symlinks"], "test_" + type + ".py")
+            os.symlink(script, symlink)
+            bc[type] = importlib.util.cache_from_source(symlink)
+
+        compileall.compile_dir(types["symlinks"], quiet=False, limit_sl_dest=types["allowed"])
+
+        self.assertTrue(os.path.isfile(bc["allowed"]))
+        self.assertFalse(os.path.isfile(bc["prohibited"]))
```

Comment:
The test is somehow hard to read (I needed to read it 3 times). It might be easier to understand it everything was repeated with explicit values instead of for-looped? I.e. sacrifice DRY for readability?

Reply:
I know what you mean. I did it this way first and then switched the test to for loops. It might be reverted to the longer and more verbose version but it would be 3 times longer. Rewrite just the second part (the second for loop) might be enough?

Reply:
I'd prefer to have it 3 times longer.

Reply:
Fixed. I've also added the exact same test but the new one is testing CLI usage. Do you like it?

Reply:
The last thing I am waiting for is your approval of the way how tests are implemented.